lib/buffer.js

```diff
@@ -67,13 +73,60 @@ function Buffer(arg, encoding) {
   return fromObject(arg);
 }
 
+/**
+ * Functionally equivalent to Buffer(arg, encoding) but throws a TypeError
+ * if value is a number.
+ **/
+Buffer.from = function(value, encoding) {
+  if (typeof value === 'number')
+    throw new TypeError(`'value' argument must not be a number`);
+  return typeof value === 'string' ?
+    fromString(value, encoding) :
+    fromObject(value);
+};
+
 Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype);
 Object.setPrototypeOf(Buffer, Uint8Array);
 
+/**
+ * Creates a new zero-filled Buffer instance.
+ **/
+Buffer.alloc = function(size, fill) {
+  if (typeof size !== 'number')
+    throw new TypeError(`'size' argument must be a number`);
+  if (fill && typeof fill !== 'string')
+    throw new TypeError(`If given, 'fill' argument must be a string`);
+  size = Math.max(0, size) || 0;
```

Comment:
I'll assume this is only to handle the case where `Number.isNaN('size')`. But I also see the same check is performed by the need to run `new Uint8Array()`. Basically, don't need the `|| 0`.